### ISSUE TITLE
Fix cmake warning popping up on ubuntu 24.04 (potentially 22) as well

### ISSUE THREAD

**Issue by badumbatish** [Shuriken-Group/Shuriken-Analyzer#104]
Take example of https://github.com/Shuriken-Group/Shuriken-Analyzer/actions/runs/12211424362/job/34068668222

where cmake warns us with 

"CMake Warning: ...." such as 
```
CMake Warning at externals/CMakeLists.txt:33 (message):
  Fetch offline mode: requires already populated _deps


CMake Warning at externals/CMakeLists.txt:40 (message):
  Running as a github action, download fetch the content


-- Cloning spdlog
CMake Warning (dev) at /usr/local/share/cmake-3.31/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(extern_spdlog) is deprecated, call
  FetchContent_MakeAvailable(extern_spdlog) instead.  Policy CMP0169 can be
  set to OLD to allow FetchContent_Populate(extern_spdlog) to be called
  directly for now, but the ability to call it with declared details will be
  removed completely in a future version.
Call Stack (most recent call first):
  externals/CMakeLists.txt:60 (FetchContent_Populate)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Populating extern_spdlog
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/runner/work/Shuriken-Analyzer/Shuriken-Analyzer/build/_deps/extern_spdlog-subbuild
[ 11%] Creating directories for 'extern_spdlog-populate'
[ 22%] Performing download step (git clone) for 'extern_spdlog-populate'
Cloning into 'extern_spdlog-src'...
HEAD is now at 7e635fca Fixed #2724 by excluding bin_to_hex sink if using std::format
[ 33%] Performing update step for 'extern_spdlog-populate'
-- Already at requested tag: v1.12.0
[ 44%] No patch step for 'extern_spdlog-populate'
[ [55](https://github.com/Shuriken-Group/Shuriken-Analyzer/actions/runs/12211424362/job/34068668222#step:11:56)%] No configure step for 'extern_spdlog-populate'
[ 66%] No build step for 'extern_spdlog-populate'
[ 77%] No install step for 'extern_spdlog-populate'
```
